WonderWoerter.py

```diff
@@ -0,0 +1,175 @@
+         self.wortSchauen.fuegeVarianteHinzu("blick", ["schaust ins Inventar und siehst ", "guckst ins Inventar und erblickst ", "blickst ins Inventar und begutachtest ","schaust in die Tasche und erkennst", "guckst in die Tasche und siehst " ,"blickst ind die Tasche und erkennst ", "schaust in den Beutel und siehst ", "guckst in den Beutel und erkennst", "blickst in den Beutel und begutachtest"])
+
+    def __initialisiereInventar(self, wortTyp:WortTyp):
+         self.wortInventar = Wort("Inventar", wortTyp)
+         self.wortInventar.fuegeVarianteHinzu("inventar", ["schaust ins Inventar und siehst ", "guckst ins Inventar und erblickst " ,"blickst ins Inventar und begutachtest ","schaust in die Tasche und erkennst", "guckst in die Tasche und siehst " ,"blickst ind die Tasche und erkennst ", "schaust in den Beutel und siehst ", "guckst in den Beutel und erkennst" ,"blickst in den Beutel und begutachtest"])
+         self.wortInventar.fuegeVarianteHinzu("tasche",["schaust ins Inventar und siehst ", "guckst ins Inventar und erblickst ", "blickst ins Inventar und begutachtest ","schaust in die Tasche und erkennst", "guckst in die Tasche und siehst ", "blickst ind die Tasche und erkennst ", "schaust in den Beutel und siehst ", "guckst in den Beutel und erkennst" ,"blickst in den Beutel und begutachtest"])
+         self.wortInventar.fuegeVarianteHinzu("beutel",["schaust ins Inventar und siehst ", "guckst ins Inventar und erblickst ", "blickst ins Inventar und begutachtest ","schaust in die Tasche und erkennst", "guckst in die Tasche und siehst ", "blickst ind die Tasche und erkennst ", "schaust in den Beutel und siehst ", "guckst in den Beutel und erkennst" ,"blickst in den Beutel und begutachtest"])
+ 
+   
+class WortGeneratoren(object):
+    def  __init__(self):
+        self.flaechenWorte =  WortGeneratorFlaeche(WortTyp.Flaeche)
+        self.ebenenWorte = WortGeneratorEbene(WortTyp.Ebene)
+        self.uebergangWorte = WortGeneratorUebergang(WortTyp.Uebergang)
+        self.blumenpflegeWorte = WortGeneratorBlumenpflege(WortTyp.Blumenpflege)
+        self.inventarWorte = WortGeneratorInventar(WortTyp.Inventar)
+        self.spielbeendenWorte =  WortGeneratorSpielBeenden(WortTyp.SpielBeenden)
+        
+    def gebeGeneratoren(self):
+        yield self.flaechenWorte
+        yield self.ebenenWorte
+        yield self.uebergangWorte    
+        yield self.blumenpflegeWorte
+        yield self.inventarWorte
+        yield self.spielbeendenWorte
+        
+class WortVergleicher(object):
+    def  __init__(self):
+         self.wortGeneratoren = WortGeneratoren()
+
+    def vergleiche(self,eingabe):
+        for WortGenerator in  self.wortGeneratoren.gebeGeneratoren():
+            for WortGen in WortGenerator.gebeWorte():
+                for Wort in WortGen.gebeAlleMoeglichenEingaben():
+                    if Wort in eingabe:
+                        return WortGen
+        return None
+    
+    def gebeWort(self, bezeichnung):
```

Comment:
Ich hab hier keinen besseren Vorschlag aber: Immer wenn du mehrfahr verschachtelte for-Schleifen (oder welche auch immer Schleifen) hast, halte inne und überlege ob man die Logik nicht anders machen kann. Wenn du mit größeren Daten arbeitest, ist eine Laufzeit von n^3 ziemlich schlecht und die Berechnung hiervon kann sehr lange dauern.